tensorflow/core/kernels/data/range_dataset_op_test.cc

```diff
@@ -0,0 +1,207 @@
+    params_.get()->runner = &runner_;
+
+    step_container_.reset(new ScopedStepContainer(0, [](const string&) {}));
+    params_->step_container = step_container_.get();
+    std::vector<AllocatorAttributes> attrs;
+    test::SetOutputAttrs(params_.get(), &attrs);
+    checkpoint::TensorSliceReaderCacheWrapper slice_reader_cache_wrapper;
+    params_.get()->slice_reader_cache = &slice_reader_cache_wrapper;
+    params_.get()->resource_manager = device_.get()->resource_manager();
+
+    context_.reset(new OpKernelContext(params_.get()));
+    device_->Compute(kernel_.get(), context_.get());
+    return context_->status();
+  }
+
+ protected:
+  template <typename T>
+  void MakeOpDef() {
+    DataType value_type = tensorflow::DataTypeToEnum<T>::value;
+    std::vector<PartialTensorShape>* shapes =
+        new std::vector<PartialTensorShape>({{}});
+    DataTypeVector* dtypes = new DataTypeVector({value_type});
+
+    TF_CHECK_OK(NodeDefBuilder("rangedataset", "RangeDataset")
+                    .Input(FakeInput(DT_INT64))
+                    .Input(FakeInput(DT_INT64))
+                    .Input(FakeInput(DT_INT64))
+                    .Attr("output_types", *dtypes)
+                    .Attr("output_shapes", *shapes)
+                    .Finalize(node_def()));
+    TF_ASSERT_OK(InitOp());
+  }
+
+  Status MakeDataset(int64 start, int64 end, int64 step, int output_index = 0,
+                     int thread_num = 2) {
+    AddInputFromArray<int64>(TensorShape({}), {start});
+    AddInputFromArray<int64>(TensorShape({}), {end});
+    AddInputFromArray<int64>(TensorShape({}), {step});
+
+    TF_CHECK_OK(InitThreadPool(thread_num));
+    TF_CHECK_OK(InitFunctionLibraryRuntime({}, thread_pool_));
+    TF_CHECK_OK(RunOpKernel());
+
+    TF_CHECK_OK(GetDatasetOutputFromContext(output_index));
+    return Status::OK();
+  }
+
+  Status MakeIteratorContext() {
+    iteratorContext_ = new IteratorContext(context_.get());
+    return Status::OK();
+  }
+
+  Status MakeIterator() {
+    iterator_.reset(nullptr);
+    TF_CHECK_OK(
+        dataset_->MakeIterator(iteratorContext_, "Iterator", &iterator_));
+    return Status::OK();
+  }
+
+  Status GetNext(std::vector<Tensor>* out_tensors, bool* end_of_sequence) {
+    TF_CHECK_OK(
+        iterator_->GetNext(iteratorContext_, out_tensors, end_of_sequence));
+    return Status::OK();
+  }
+
+ protected:
+  std::unique_ptr<DeviceMgr> device_mgr_;
+  FunctionLibraryRuntime* flr_;
+  std::unique_ptr<FunctionLibraryDefinition> lib_def_;
+  std::unique_ptr<ProcessFunctionLibraryRuntime> pflr_;
+  FunctionDefLibrary fdef_lib_;
+  thread::ThreadPool* thread_pool_;
+  std::function<void(std::function<void()>)> runner_;
+  DatasetBase* dataset_;
+  IteratorContext* iteratorContext_;
+  std::unique_ptr<IteratorBase> iterator_;
+};
+
+TEST_F(RangeDatasetOpTest, GetNext) {
+  MakeOpDef<int64>();
+  int start = 0, end = 10, step = 1;
+  TF_CHECK_OK(MakeDataset(start, end, step));
```

Comment:
in tests, use TF_ASSERT_OK for things that are not expected to fail and TF_EXPECT_OK for expectations